sway/layout.c

```diff
@@ -65,6 +65,15 @@ swayc_t *replace_child(swayc_t *child, swayc_t *new_child) {
 
 swayc_t *remove_child(swayc_t *parent, swayc_t *child) {
 	int i;
+	// Special case for floating views
+	if (child->is_floating) {
+		for (i = 0; i < parent->floating->length; ++i) {
+			if (parent->floating->items[i] == child) {
+				list_del(parent->floating, i);
+				break;
+			}
+		}
+	}
```

Comment:
The if statement isn't necessary, but it saves an O(n) look up at the cost of that O(1) check. Seems like a good tradeoff to me.